### PR TITLE
[RayCluster] Adds `aliyun.com/gpu-mem` to known custom accelerators

### LINKED ISSUE #2484: Statistics of other types of gpu
### Search before asking

- [X] I had searched in the [issues](https://github.com/ray-project/kuberay/issues) and found no similar feature requirement.

### Description

https://github.com/ray-project/kuberay/blob/33ba38546ddda0fa9121df20b357a13d47bb90d9/ray-operator/controllers/ray/raycluster_controller.go#L1623

if use aliyun k8s gpu share, gpu key is `aliyun.com/gpu-mem`
```
    workerGroupSpecs:
            resources:
              limits:
                aliyun.com/gpu-mem: "1"
                cpu: "1"
                memory: 2Gi
              requests:
                aliyun.com/gpu-mem: "1"
                cpu: "1"
                memory: 2Gi
```

autoscaler will not work when request gpu resource
```
(autoscaler +3m13s) Error: No available node types can fulfill resource request {'GPU': 1.0, 'CPU': 1.0}. Add suitable node types to this cluster to resolve this issue.
```

code:
```python
import ray
import time

ray.init()

@ray.remote(num_gpus=1) 
def gpu_task():
    import torch
    x = torch.rand(10000, 10000).cuda()  
    y = torch.mm(x, x) 
    return y.sum().item()

future = gpu_task.remote()
result = ray.get(future)

print("Result:", result)

ray.shutdown()
```

### Use case

_No response_

### Related issues

none

### Are you willing to submit a PR?

- [ ] Yes I am willing to submit a PR!

### PR DESCRIPTION
**Pull request by win5923**
## Why are these changes needed?

if use aliyun k8s gpu share, gpu key is `aliyun.com/gpu-mem`
```yaml
    workerGroupSpecs:
            resources:
              limits:
                aliyun.com/gpu-mem: "1"
                cpu: "1"
                memory: 2Gi
              requests:
                aliyun.com/gpu-mem: "1"
                cpu: "1"
                memory: 2Gi
```
autoscaler will not work when request gpu resource
```
(autoscaler +3m13s) Error: No available node types can fulfill resource request {'GPU': 1.0, 'CPU': 1.0}. Add suitable node types to this cluster to resolve this issue.
```

Adds aliyun's gpu share to the list of known custom accelerators in `pod.go`

## Related issue number

Closes #2484

## Checks

- [x] I've made sure the tests are passing.
- Testing Strategy
   - [x] Unit tests
   - [ ] Manual tests
   - [ ] This PR is not tested :(

### CODE REVIEW

**Comment by win5923** [comment]
I'm curious why the testing for `HeadNode with multiple custom accelerators` lack `"google.com/tpu"` values?

https://github.com/ray-project/kuberay/blob/073de1f62bb9e912e4f0a5ab318ab8bdcce298a4/ray-operator/controllers/ray/common/pod_test.go#L1270-L1282

File: ray-operator/controllers/ray/common/pod.go [hunk 1]
```diff
@@ -38,11 +38,14 @@ const (
 	NeuronCoreRayResourceName          = "neuron_cores"
 	TPUContainerResourceName           = "google.com/tpu"
 	TPURayResourceName                 = "TPU"
+	GPUShareContainerResourceName      = "aliyun.com/gpu-mem"
+	GPUShareResourceName               = "gpu_share"
```
Comment (andrewsykim) [inline ray-operator/controllers/ray/common/pod.go:42]: Is this the actual resource name recognized by Ray? `aliyun.com/gpu-mem` -> `gpu_share` seems like an odd mapping

Comment (win5923) [reply ray-operator/controllers/ray/common/pod.go:42]: I think this is likely a custom mapping.

Comment (win5923) [reply ray-operator/controllers/ray/common/pod.go:42]: Maybe naming it `aliyun-gpu` would be better?

Comment (win5923) [reply ray-operator/controllers/ray/common/pod.go:42]: Apologies, I realized that Ray Resource Name only supports GPU, neuron_cores, TPU, NPU, and HPU. I will mapping aliyun's gpu shae to GPU.

Comment (andrewsykim) [reply ray-operator/controllers/ray/common/pod.go:42]: yes, I think we should only add custom accelerators here that are in the supported list you referenced

**Comment by win5923** [comment]
cc @zjj2wry

**Review by andrewsykim — CHANGES_REQUESTED** [review]
ref: https://github.com/ray-project/kuberay/pull/2631#discussion_r1878545981

File: ray-operator/controllers/ray/common/pod.go (hunk 1 above)
Comment (win5923) [reply ray-operator/controllers/ray/common/pod.go]: I mapped `aliyun.com/gpu-mem` to `num-gpus`.

**Comment by win5923** [comment]
I mapped `aliyun.com/gpu-mem` to `num-gpus`. Because gpu share also belongs to the GPU.

**Comment by andrewsykim** [comment]
Can you first update a PR to update the officially supported accelerators in https://docs.ray.io/en/latest/ray-core/scheduling/accelerators.html to include alyun.com/gpu?

**Comment by win5923** [comment]
Hi @andrewsykim, I checked Aliyun's documentation, and I think they also use Nvidia GPUs. Do we need to add a custom accelerator for this?

Ref: https://www.alibabacloud.com/help/en/eci/user-guide/create-a-pod-by-specifying-the-gpu-specification-1

**Comment by win5923** [comment]
I checked Aliyun's documentation, and I think they also use Nvidia GPU with field called `nvidia.com/gpu`, while Aliyun GPU Share has another field called `aliyun.com/gpu-mem`.

Ref: https://www.alibabacloud.com/help/en/eci/user-guide/create-a-pod-by-specifying-the-gpu-specification-1

> Can you first open a PR to update the officially supported accelerators in https://docs.ray.io/en/latest/ray-core/scheduling/accelerators.html to include alyun.com/gpu?

So I think we should just focus on the label?
